src/object.c

```diff
@@ -1471,40 +2336,65 @@ static rct_string_id object_type_park_entrance_desc(void *objectEntry)
 	return STR_NONE;
 }
 
+static void object_type_park_entrance_reset(void *objectEntry, uint32 entryIndex)
+{
+	rct_entrance_type *entranceType = (rct_entrance_type*)objectEntry;
+	uint8 *extendedEntryData = (uint8*)((size_t)objectEntry + sizeof(rct_entrance_type));
+
+	entranceType->string_idx = object_get_localised_text(&extendedEntryData, OBJECT_TYPE_PARK_ENTRANCE, entryIndex, 0);
+	entranceType->image_id = object_chunk_load_image_directory(&extendedEntryData);
+
+	uintptr_t some_pointer = RCT2_GLOBAL(0x9ADAF4, uint32);
+	if (some_pointer != (uintptr_t)0xFFFFFFFF){
+		*((uint16*)some_pointer) = 0;
+	}
+}
+
 static const object_type_vtable object_type_park_entrance_vtable[] = {
 	object_type_park_entrance_load,
 	object_type_park_entrance_unload,
 	object_type_park_entrance_test,
 	object_type_park_entrance_paint,
-	object_type_park_entrance_desc
+	object_type_park_entrance_desc,
+	object_type_park_entrance_reset,
 };
 
 ///////////////////////////////////////////////////////////////////////////////
 // Water (rct2: 0x006E6E2A)
 ///////////////////////////////////////////////////////////////////////////////
 
-static bool object_type_water_load(void *objectEntry, uint32 entryIndex)
+static uint8* object_type_water_load(void *objectEntry, uint32 entryIndex, int *chunkSize)
 {
 	rct_water_type *waterEntry = (rct_water_type*)objectEntry;
+	const uint8 *origExtendedEntryData = (uint8*)((size_t)objectEntry + sizeof(rct_water_type));
+	const size_t extendedDataSize = *chunkSize - sizeof(rct_water_type);
+	*chunkSize = *chunkSize + sizeof(rct_water_type) - sizeof(rct_water_type);
```

Comment:
There are a lot of other types which don't quite match and they have to use such things to calculate proper offsets. When I was doing this change I initially changed all of these to have such changes for consistency and only changed ones that required it.

Even if some of these structs may not have pointers in them, it would be nice to be able to load them into copies that are free of `#pragma pack(push, 1)`.